samples/wcf-relay/RoleBasedAccessControl/Program.cs

```diff
@@ -0,0 +1,338 @@
+                // Includes NetEvent as well
+                CreateAndSendWithNotificationChannel(address, tokenProvider, binding);
+            }
+            else if (binding is WebHttpBinding)
+            {
+                CreateAndSendWithWebRequestResponseChannel(address, tokenProvider, binding);
+            }
+        }
+
+        static void CreateAndSendWithMathChannel(Uri address, TokenProvider tokenProvider, Binding binding)
+        {
+            ChannelFactory<IMathClient> channelFactory = null;
+            IMathClient client = null;
+
+            try
+            {
+                channelFactory = new ChannelFactory<IMathClient>(binding, address.AbsoluteUri);
+                channelFactory.Endpoint.Behaviors.Add(new TransportClientEndpointBehavior(tokenProvider));
+                client = channelFactory.CreateChannel();
+                client.Add(1, 1);
+            }
+            finally
+            {
+                SafeClose(client);
+                SafeClose(channelFactory);
+            }
+        }
+
+        static void CreateAndSendWithNotificationChannel(Uri address, TokenProvider tokenProvider, Binding binding)
+        {
+            ChannelFactory<INotificationService> channelFactory = null;
+            INotificationService channel = null;
+
+            try
+            {
+                channelFactory = new ChannelFactory<INotificationService>(binding, address.AbsoluteUri);
+                channelFactory.Endpoint.Behaviors.Add(new TransportClientEndpointBehavior(tokenProvider));
+                channel = channelFactory.CreateChannel();
+                channel.Notify(Guid.NewGuid().ToString(), "My event");
+            }
+            finally
+            {
+                SafeClose((ICommunicationObject)channel);
+                SafeClose(channelFactory);
+            }
+        }
+
+        static void CreateAndSendWithWebRequestResponseChannel(Uri address, TokenProvider tokenProvider, Binding binding)
+        {
+            var channelFactory = new ChannelFactory<IWebRequestResponseChannel>(binding, new EndpointAddress(address));
+            IWebRequestResponseChannel client = null;
+
+            try
+            {
+                channelFactory.Endpoint.Behaviors.Add(new TransportClientEndpointBehavior(tokenProvider));
+                channelFactory.Endpoint.Behaviors.Add(new WebHttpBehavior());
+                client = channelFactory.CreateChannel();
+                string getResponse = client.SimpleGetString();
+                string postResponse = client.SimplePostString("my string");
+                string getArgsResponse = client.GetWithUriArgs("arg1", "arg2");
+            }
+            finally
+            {
+                SafeClose(client);
+                SafeClose(channelFactory);
+            }
+        }
+
+        /***********************************************************************
+        * The methods below are not related to Role Based Access Control. 
```

Comment:
>The methods below are not related to Role Based Access Control.  [](http://example.com/codeflow?start=10&length=64)

Arguably everthing from line 121 and below is code that is not RBAC dependent -- it works with SAS or RBAC.